### PR TITLE
volmesh class update

### PR DESCRIPTION
**Pull request by juney-lee**
- various updates and clean up of the volmesh class (ongoing and more to come...)
- halffaces are restructured as list of vertex keys (like the faces of a mesh)

### CODE REVIEW

**Comment by gonzalocasas** [comment]
@juney-lee if possible, please don't delete the template of the PR, and instead use it to tick the type of change you're suggesting. Also, pls add an entry to the `CHANGELOG.md` file ;)

File: src/compas/datastructures/volmesh/volmesh.py [hunk 1]
```diff
@@ -459,8 +586,10 @@ def add_halfface(self, vertices, fkey=None):
             raise Exception('Corrupt halfface.')
 
         fkey = self._get_face_key(fkey)
+        self.halfface[fkey] = vertices
+        self.facedata[fkey] = self.default_face_attributes
 
-        self.halfface[fkey] = {}
+        edge_attr = self.default_edge_attributes
```
Comment (tomvanmele) [inline src/compas/datastructures/volmesh/volmesh.py:592]: this should be a copy

File: src/compas/datastructures/volmesh/volmesh.py [hunk 2]
```diff
@@ -567,19 +729,8 @@ def edges(self, data=False):
...
-        #         vertices = self.halfface_vertices(fkey, ordered=True)
-        #         vset = frozenset(vertices)
-        #         if vset not in seen:
-        #             faces.append(vertices)
-        #         seen.add(vset)
-        # return faces
+        """"Return unique" halfface keys."""
         seen = set()
         faces = []
         for fkey in self.halfface:
```
Comment (tomvanmele) [inline src/compas/datastructures/volmesh/volmesh.py:736]: should this not be `for fkey in self.halffaces()`

Comment (tomvanmele) [reply src/compas/datastructures/volmesh/volmesh.py:736]: is there such a thing as `self.halffaces()`?

**Review by tomvanmele — CHANGES_REQUESTED** [review]
the implementation should be equivalent with the other data structures.
* use iterators for all accessor functions
* all data should be accessible through methods

**Pull request closed without merging**